src/main/java/com/google/sps/lib/Graph.java

```diff
@@ -21,47 +21,31 @@
  */
 public class Graph {
   private final List<Task> taskList;
-  private final int listSize;
   private boolean isGraphCreated;
   private double[][] distanceMatrix;
-  ICalculateDistance distance;
+  IDistanceCalculator distanceCalculator;
+  ITSPAlgorithm tspAlgorithm;
 
   public Graph(List<Task> taskList) {
     this.taskList = taskList;
-    this.listSize = taskList.size();
     this.isGraphCreated = false;
+    this.distanceCalculator = new EuclideanDistanceCalculator();
   }
 
-  /**
-   * Getter for distanceMatrix.
-   */
-  public double[][] getGraph() {
+  public List<String> getMinimumPath() {
     if (!this.isGraphCreated) {
-      this.distanceMatrix = createGraph();
+      this.distanceMatrix = calculateDistance();
       this.isGraphCreated = true;
     }
-    return this.distanceMatrix;
-  }
-
-  private void setCalculateDistanceMethod(ICalculateDistance distance) {
-    this.distance = distance;
-  }
-
-  private double calculateDistance(int i, int j) {
-    setCalculateDistanceMethod(new calculateDistanceUsingEuclidean());
-    return distance.findDistance(taskList, i, j);
+    if(taskList.size() <= 15) {
```

Comment:
Why this value is hardcoded? Please move this value to a private static final int.